docs/user/install-gke-external.md

```diff
@@ -0,0 +1,246 @@
+3. Select the `Desktop or Native application` application type.
+4. Fill in an appropiate application name.
+5. Please make a note of the generated `Client ID` and `Client Secret` for later use.
+
+You will then need to generate a cookie secret for later use:
+
+```
+python3 -c 'import os,base64; print(base64.urlsafe_b64encode(os.urandom(32)).decode())'
+```
+
+With the Turbinia repository cloned to your local machine, cd into the directory we'll be working from:
+
+```
+wyassine@wyassine:~/turbinia$ cd k8s/tools/
+```
+
+Then make a copy of the `oauth2_proxy.cfg` template:
+
+```
+wyassine@wyassine:~/turbinia/k8s/tools$ cp ../../docker/oauth2_proxy/oauth2_proxy.cfg .
+```
+
+Edit the `oauth2_proxy.cfg` file and replace the following:
+
+- `CLIENT_ID`: The web client id
+- `CLIENT_SECRET`: The web client secret
+- `OIDC_EXTRA_AUDIENCES`: The native client id
+- `UPSTREAMS`: The domain name registered, ex: `upstreams = ['https://<DOMAIN>]`
+- `REDIRECT_URL`: The redirect URI you registered ex: `https://<DOMAIN>/oauth2/callback`
+- `COOKIE_SECRET`: The cookie secret you generated above
+- `EMAIl_DOMAINS`: The email domain name you'd allow to authenticate ex: `yourcompany.com`
+
+Now base64 encode the config file:
+
+```
+base64 -w0 oauth2_native.cfg > oauth2_native.b64
+```
+
+Then to deploy the config to the cluster
+
+```
+kubectl create configmap oauth2-config --from-file=OAUTH2_CONF=oauth2_native.b64
+```
+
+Create a file named `auth.txt` in your working directory and append a list of emails
+you'd like to allow access to the Turbinia app, one email per line. Once complete base64 encode:
+
+```
+base64 -w0 auth.txt > auth.b64
+```
+
+Then deploy the config to the cluster:
+
+```
+kubectl create configmap auth-config --from-file=OAUTH2_AUTH_EMAILS=auth.b64
+```
+
+Lastly, deploy the Oauth2 Proxy to the cluster:
+
+```
+kubectl create -f ../celery/turbinia-oauth2-proxy.yaml
+```
+
+### 4. Deploy the Load Balancer and Managed SSL
+
+In the final step, edit `turbinia-ingress.yaml` located in the `k8s/celery` directory
+and replace the two placeholders `<DOMAIN_NAME>` with the domain you configured. Save
+the file then deploy it to the cluster:
+
+```
+kubectl create -f ../celery/turbinia-ingress.yaml
+```
+
+Within 10 minutes all the load balancer components should be ready and you should
+be able to externally connect to the domain name you configured. Additionally, you can check on the status of the load balancer via:
+
+```
+kubectl describe ingress turbinia-ingress
+```
+
+Congrats, you have now successfully configured Turbinia to be externally accessible!
+
+## Making Turbinia processing requests
+
+Once Turbinia is externally accessible, download the Oauth Desktop credentials
+created above to your machine and install the Turbinia client:
+
+```
+pip3 install turbinia_api_client`
+```
+
+- To create a processing request for evidence run the following:
+
+```
+turbiniamgmt submit googleclouddisk -d <DISK_NAME> -z <ZONE>
```

Comment:
An example of supported options for googleclouddisk might be helpful here.

```
Usage: turbiniamgmt.py submit GoogleCloudDisk [OPTIONS]

Options:
  --project TEXT          [default: (project); required]
  --zone TEXT             [default: (zone); required]
  --disk_name TEXT        [default: (disk_name); required]
  --mount_partition TEXT  [default: (mount_partition)]
  --source TEXT           [default: (source)]
  --name TEXT             [default: (name)]
  --description TEXT      [default: (description)]
  --tags TEXT             [default: (tags)]
  --recipe_name TEXT      [default: (recipe_name)]
  --recipe_data TEXT      [default: (recipe_data)]
  --jobs_allowlist TEXT   [default: (jobs_allowlist)]
  --jobs_denylist TEXT    [default: (jobs_denylist)]
  --filter_patterns TEXT  [default: (filter_patterns)]
  --yara_rules TEXT       [default: (yara_rules)]
  --sketch_id TEXT        [default: (sketch_id)]
  -h, --help              Show this message and exit.
```

Reply:
```turbiniamgmt submit GoogleCloudDisk --project <PROJECT_NAME> --disk_name <DISK_NAME> --zone <ZONE>```

Reply:
confirm OOB that it does need to be spelled `GoogleCloudDisk` went ahead and updated docs to represent this way and can revisit later if we want to lowercase instead